dialogue-core/src/test/java/com/palantir/dialogue/core/DialogueChannelsTest.java

```diff
@@ -155,6 +163,35 @@ public void bad_channel_throwing_an_error_still_returns_a_future() {
         assertThatThrownBy(() -> Futures.getUnchecked(future)).hasCauseInstanceOf(NoClassDefFoundError.class);
     }
 
+    @Test
+    @SuppressWarnings("FutureReturnValueIgnored") // intentionally spawning a bunch of throwaway requests
+    void live_reloading_an_extra_uri_allows_queued_requests_to_make_progress() {
+        when(delegate.execute(any(), any())).thenReturn(SettableFuture.create());
+
+        channel = DialogueChannel.builder()
+                .channelName("my-channel")
+                .clientConfiguration(stubConfig)
+                .channelFactory(uri -> delegate)
+                .build();
+
+        int numRequests = 100; // we kick off a bunch of requests but don't bother waiting for their futures.
+        for (int i = 0; i < numRequests; i++) {
+            channel.execute(endpoint, request);
+        }
+
+        assertThat(queuedRequestsCounter())
+                .describedAs("stubConfig has one uri and ConcurrencyLimitedChannel's initialLimit is 20, so there "
+                        + "should be 80 queued up requests")
+                .isEqualTo(numRequests - ConcurrencyLimitedChannel.INITIAL_LIMIT);
+
+        // live-reload from 0 -> 1
+        channel.updateUris(ImmutableList.of(stubConfig.uris().get(0), "https://some-other-uri"));
+
+        assertThat(queuedRequestsCounter())
+                .describedAs("Now that we have two uris, another batch of 20 requests can get out the door")
+                .isEqualTo(numRequests - 2 * ConcurrencyLimitedChannel.INITIAL_LIMIT);
+    }
```

Comment:
slick test :-)

Reply:
Honestly it does feel a bit weird using metrics to make assertions about behaviour rather than direct java method calls, but I guess if our metrics no longer reflect the ground truth then we have bigger problems!

Reply:
Can we assert on the number of invocations of the delegate channel instead?